### PR TITLE
II-137 Collection Page Tests

### PR DESCRIPTION
**Pull request by jinyoungkim927**
[Jira Ticket: ](https://immutable.atlassian.net/jira/software/projects/II/boards/294?selectedIssue=II-137)

The target is to improve each metric by 90%. 

Previous testing: 
Screen Shot 2022-08-11 at 3 49 36 pm
Screen Shot 2022-08-11 at 3 49 44 pm

Current testing: 
Screen Shot 2022-08-11 at 3 49 39 pm
Screen Shot 2022-08-11 at 3 49 47 pm

### CODE REVIEW

File: src/components/NFTCollection/CollectionItem.test.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+            <CollectionItem
+                name={testProps.name}
+                url={testProps.url}
+                price={testProps.price}
+                orderId={testProps.orderId} />
+        );
+        expect(container).toMatchSnapshot()
+    })
+
+    test('Load placeholder image on error', () => {
```
Comment (maksim-tsvetkov) [inline src/components/NFTCollection/CollectionItem.test.tsx:27]: the naming of test case should be consistent with others. 
```suggestion
    test('load placeholder image on error', () => {
```

File: src/components/NFTCollection/CollectionItem.test.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+                url={testProps.url}
+                price={testProps.price}
+                orderId={testProps.orderId} />
+        );
+        const image = screen.getByRole('img');
+        fireEvent.error(image)
+        expect(image).toHaveAttribute('src','placeholderImg.jpg');
+    })
+
+    test('buyItem errors correctly', async () => {
```
Comment (maksim-tsvetkov) [inline src/components/NFTCollection/CollectionItem.test.tsx:40]: not clear what you meant

File: src/pages/CollectionPage/collectionPage.test.tsx [hunk 3]
```diff
@@ -34,4 +44,21 @@ describe('Collection page', () => {
...
+
+        const collectionCard_Img = screen.getByTestId('CollectionCard__Img');
+        expect(collectionCard_Img).toHaveAttribute('src', 'placeholderImg.jpg');
+    })
+
 })
+
+
+
+// <CollectionCard name={item.name} url={item.collection_image_url || placeholderImg} />
```
Comment (maksim-tsvetkov) [inline src/pages/CollectionPage/collectionPage.test.tsx:64]: ?! 🤨

File: src/components/NFTCollection/CollectionItem.test.tsx (hunk 2 above)
Comment (jinyoungkim927) [reply src/components/NFTCollection/CollectionItem.test.tsx:40]: Yep, I meant to say: 'trigger console log if buyItem times out'

**Review by maksim-tsvetkov — APPROVED** [review]

File: src/components/NFTCollection/CollectionCard.test.tsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import { render, fireEvent, screen } from '@testing-library/react'
+import { MemoryRouter } from 'react-router-dom';
+import CollectionCard from './CollectionCard'
+import placeholderImg from '../../asset/placeholderImg.jpg'
+
+
+const testProps = {
```
Comment (ABashkirtseva) [inline src/components/NFTCollection/CollectionCard.test.tsx:7]: Better to name collectionCardProps

File: src/components/NFTCollection/CollectionCard.test.tsx [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    url: 'http.test.com',
+    name: 'collectionName'
+}
+
+describe('CollectionCard component', () => {
+    test('match snapshot', () => {
+        const { container } = render(
+            <CollectionCard
+                name={testProps.name}
+                url={testProps.url} />
```
Comment (ABashkirtseva) [inline src/components/NFTCollection/CollectionCard.test.tsx:17]: you can just use ...testProps,

File: src/components/NFTCollection/CollectionCard.test.tsx [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+    test('Load placeholder image on error', () => {
+        render(
+            <CollectionCard
+                name={testProps.name}
+                url={testProps.url} />
+        );
+        const image = screen.getByRole('img');
+        fireEvent.error(image)
+        expect(image).toHaveAttribute('src','placeholderImg.jpg');
```
Comment (ABashkirtseva) [inline src/components/NFTCollection/CollectionCard.test.tsx:30]: space after comma

File: src/components/NFTCollection/CollectionItem.test.tsx [hunk 7]
```diff
@@ -0,0 +1,56 @@
+import { render, fireEvent, screen  } from '@testing-library/react'
+import { Link } from '@imtbl/imx-sdk'
+import CollectionItem from './CollectionItem'
+
+const testProps = {
```
Comment (ABashkirtseva) [inline src/components/NFTCollection/CollectionItem.test.tsx:5]: here as well, you need to specify props of what

**Review by charlie-chau — APPROVED** [review]

**Pull request merged**